deploy/src/main/deploy/kubernetes/olm_deploy.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#*******************************************************************************
+# Copyright (c) 2018 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0
+#
+# SPDX-License-Identifier: EPL-2.0
+#*******************************************************************************
+
+set -e
+
+SCRIPTPATH="$(cd "$(dirname "$0")" && pwd -P)"
+source "$SCRIPTPATH/common.sh"
+
+echo DEPLOYING OPERATOR LIFECYCLE MANAGER TO KUBERNETES
+
+if [ $(is_olm_deployed) == "YES" ]; then
+	echo "OLM already deployed. Exiting ..."
+	exit 0
+fi
+
+
+echo
+echo "Deploying OLM ..."
+test -d operator-lifecycle-manager || git clone https://github.com/operator-framework/operator-lifecycle-manager.git
+kubectl create -f operator-lifecycle-manager/deploy/upstream/manifests/latest/ || true
+
+#
+# The next delay is necessary due to operator-framework/operator-lifecycle-manager#558
+# Sometimes it happens that in the previous step the use of a newly deployed
+# CRD fails, and so a catalog source is missing. The next steps wait "a bit"
+# and then re-register the catalog source. Which again might fail, because
+# the previous step might have been successful.
+#
+
+sleep 5 # delay and retry CRD
+
+kubectl create -f operator-lifecycle-manager/deploy/upstream/manifests/latest/*rh-operators.catalogsource.yaml || true
+
+echo ... done
```

Comment:
what about removing the local olm workspace cloned from GitHub? Shouldn't we remove it as well?

Reply:
I think it is worth keeping the content. As re-downloading is might be an issue for someone. However I did add a short message in the "undeploy" script:

~~~
➜  kubernetes git:(feature/olm_fixup_1) ✗ ./olm_undeploy.sh 
UNDEPLOYING OPERATOR LIFECYCLE MANAGER FROM KUBERNETES
namespace "olm" deleted
namespace "operators" deleted
…
OPERATOR LIFECYCLE MANAGER UNDEPLOYED FROM KUBERNETES

The git repository of OLM is still kept in the local directory. If you would like to delete it as well, you can simply run:

    rm -Rf operator-lifecycle-manager

➜  kubernetes git:(feature/olm_fixup_1) ✗
~~~